internal/github/client.go

```diff
@@ -0,0 +1,117 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// GitHubClient is a client for interacting with the GitHub API.
+type GitHubClient struct {
+	client    *http.Client
+	authToken string // GitHub Personal Access Token
+	baseURL   string // Base URL for the GitHub API
+}
+
+// NewGitHubClient creates a new GitHubClient.
+// It now takes the GitHub Personal Access Token and the base URL as arguments.
+func NewGitHubClient(baseURL string) *GitHubClient {
+	if baseURL == "" {
+		baseURL = "https://api.github.com" // Default base URL
+	}
+	return &GitHubClient{
+		client:  &http.Client{},
```

Comment:
The best practice is to always set a timeout in http clients. Otherwise your app will lock up if in some networking situations.

```suggestion
		client:  &http.Client{Timeout: time.Second *3},
```